crates/sui-faucet/src/main.rs

```diff
@@ -154,6 +156,16 @@ async fn request_gas(
     info!(uuid = ?id, "Got new gas request.");
     let result = match payload {
         FaucetRequest::FixedAmountRequest(requests) => {
+            // Assert that the request address is a valid address
+            if !requests.recipient.is_valid_address() {
+                return (
+                    StatusCode::BAD_REQUEST,
+                    Json(FaucetResponse::from(FaucetError::InvalidAddress(
+                        requests.recipient.to_string(),
+                    ))),
+                );
+            };
```

Comment:
```suggestion
```
I don't think this can is the right place to add. The data should have been rejected on line 152 already if it is not valid, which is why we need to check before that step